packages/gatsby/cache-dir/loader.js

```diff
@@ -226,6 +230,54 @@ export class BaseLoader {
           // undefined if final result is an error
           return pageResources
         })
+
+        const staticQueryBatchPromise = Promise.all(
+          staticQueryHashes.map(staticQueryHash => {
+            console.log(
+              `${pagePath} needs a static query with hash ${staticQueryHash}`
+            )
+
+            // TODO: Get from cache
+
+            // TODO: Get in flight Promise
+
+            return doFetch(`/static/d/${staticQueryHash}.json`).then(req => {
+              const jsonPayload = JSON.parse(req.responseText)
+              console.log(
+                `Static query with hash ${staticQueryHash} was fetched`,
+                jsonPayload
+              )
+              // this.staticQueryDb.set(staticQueryHash, jsonPayload)
+              return { staticQueryHash, jsonPayload }
+            })
+          })
+        ).then(staticQueryResults => {
+          console.log({
+            staticQueryResults,
+          })
+          const staticQueryResultsMap = staticQueryResults.reduce(
+            (map, { staticQueryHash, jsonPayload }) => {
+              map[staticQueryHash] = jsonPayload
+              return map
+            },
+            {}
```

Comment:
Mandatory "reduce" suggestion here
```suggestion
          const staticQueryResultsMap = {}
          staticQueryResults.forEach(
            ({ staticQueryHash, jsonPayload }) =>
              staticQueryResultsMap[staticQueryHash] = jsonPayload
```

Reply:
Done